app/src/main/java/com/epfl/drawyourpath/community/TournamentModel.kt

```diff
@@ -1,51 +1,249 @@
 package com.epfl.drawyourpath.community
 
+import android.content.Context
+import android.widget.Toast
+import androidx.lifecycle.*
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.epfl.drawyourpath.database.Database
+import com.epfl.drawyourpath.database.MockDatabase
+import com.epfl.drawyourpath.path.Run
+import java.time.LocalDateTime
+import java.util.concurrent.CompletableFuture
+
 /**
- *
+ * this class is used to link the database to the UI for the tournaments
  */
-class TournamentModel : java.io.Serializable {
+class TournamentModel(
+    val database: Database,
+    private var currentUserId: String,
+) : ViewModel() {
+    private val _userTournaments: MutableList<String> = MockDatabase.mockUser.tournaments!!.toMutableList()
+
+    private val userTournaments: MutableLiveData<List<String>> = MutableLiveData(_userTournaments)
+
+    private val allTournamentIds = database.getAllTournamentsId()
+
+    private val allTournamentInfo: MutableMap<String, LiveData<Tournament>> = mutableMapOf()
+
+    private val allTournaments: LiveData<List<Tournament>> = MediatorLiveData<List<Tournament>>().apply {
+        addSource(allTournamentIds) { tournamentIds ->

```

Comment:
Maybe you can refactor this functions with some helpers functions because therere a lot of block inside block, or add documentations